PhysicsTools/PatExamples/plugins/MuonVIDCITKAnalyzer.cc

```diff
@@ -0,0 +1,206 @@
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDAnalyzer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/Event.h"
+
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "DataFormats/Common/interface/View.h"
+#include "DataFormats/Common/interface/ValueMap.h"
+#include "DataFormats/MuonReco/interface/Muon.h"
+#include "DataFormats/MuonReco/interface/MuonFwd.h"
+#include "DataFormats/MuonReco/interface/MuonSelectors.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+
+#include "TH1F.h"
+#include "TH2F.h"
+
+#include <sstream>
+
+using namespace std;
+using namespace edm;
+//
+// class decleration
+//
+class MuonVIDCITKAnalyzer : public edm::EDAnalyzer 
```

Comment:
Please make your code thread-safe. You can check the static analyzer [report](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9273/5027/llvm-analysis/) for the instances of unsafe code.